flink-table/flink-table-planner/src/main/java/org/apache/flink/table/planner/operations/MergeTableLikeUtil.java

```diff
@@ -494,7 +494,13 @@ private void collectPhysicalFieldsTypes(List<SqlNode> derivedColumns) {
                     boolean nullable = type.getNullable() == null ? true : type.getNullable();
                     RelDataType relType = type.deriveType(sqlValidator, nullable);
                     // add field name and field type to physical field list
-                    physicalFieldNamesToTypes.put(name, relType);
+                    RelDataType oldType = physicalFieldNamesToTypes.put(name, relType);
```

Comment:
I think it is not enough to add check here, when there is name conflicts between computedColumn or metadata Column, the check here would not work well. you can try to add validation in appendDerivedColumns.

Reply:
> I think it is not enough to add check here, when there is name conflicts between computedColumn or metadata Column, the check here would not work well. you can try to add validation in appendDerivedColumns.

Hi, @wenlong88 Thanks for your advice. I think it is very meaningful. I will verify it first and make adjustments in time if there is any problem

Reply:
> I think it is not enough to add check here, when there is name conflicts between computedColumn or metadata Column, the check here would not work well. you can try to add validation in appendDerivedColumns.

Hi, @wenlong88  Thanks for your advice, which is of great help to me. I found several problems during the testing process
1. Duplicate columns are overwritten when the compute column is first and the regular column is last
2. If the metadata column is first, use the compute column in the middle and the regular column in the rear. If the metadata column name is the same as the regular column name, use the compute column in the middle and the regular column name. In the generated comput column process, namely in accessibleFieldNamesToTypes. PutAll methods covered in repeated fields.

I have adjusted the code accordingly and added test for adjustment. Please help to review it